### ISSUE TITLE
Panic runtime error for 0.4.0 in GKE 1.5.2

### ISSUE THREAD

**Issue by rvrignaud** [kubernetes/kube-state-metrics#84]
Hello,

I just tried to upgrade kube-state-metric pod in 0.4.0. I'm running a GKE cluster in 1.5.2.
Kube-state-metrics is running in Ubuntu 16.04 container and built using:
```
apt-get install -y golang git
cd /root
mkdir -p "$GOPATH"
go get github.com/kubernetes/kube-state-metrics
mv "${GOPATH}/bin/kube-state-metrics" /usr/local/bin/
```

Here is the stacktrace I got when I try to get /metric

```
kube-state-metrics
I0210 07:30:30.326926     175 main.go:139] Using default collectors
I0210 07:30:30.327196     175 main.go:186] service account token present: true
I0210 07:30:30.327224     175 main.go:187] service host: https://10.255.240.1:443
I0210 07:30:30.327902     175 main.go:213] Testing communication with server
I0210 07:30:30.359162     175 main.go:218] Communication with server successful
I0210 07:30:30.359497     175 main.go:263] Active collectors: resourcequotas,replicasets,daemonsets,deployments,pods,nodes
I0210 07:30:30.359533     175 main.go:227] Starting metrics server: :80
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x402578]
goroutine 81 [running]:
panic(0x18c3a00, 0xc82000a0a0)
	/usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
main.(*deploymentCollector).collectDeployment(0xc820456490, 0xc82006e900, 0x0, 0x0, 0x0, 0x0, 0xc820495bc4, 0x6, 0x0, 0x0, ...)
	/root/go/src/github.com/kubernetes/kube-state-metrics/deployment.go:153 +0x238
main.(*deploymentCollector).Collect(0xc820456490, 0xc82006e900)
	/root/go/src/github.com/kubernetes/kube-state-metrics/deployment.go:135 +0x232
github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc8204842c0, 0xc82006e900, 0x7f4865a756a8, 0xc820456490)
	/root/go/src/github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:433 +0x58
created by github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/root/go/src/github.com/kubernetes/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:434 +0x360
```

**Comment by brancz** [comment]
To perform some testing myself in the mean time, until the new image is published on gcr I have an image at `quay.io/brancz/kube-state-metrics:v0.4.0`, could you try that image?

**Comment by rvrignaud** [comment]
Hello @brancz ,

I have same behavior with you image:
```
I0210 16:30:55.774545       1 main.go:139] Using default collectors
I0210 16:30:55.774749       1 main.go:186] service account token present: true
I0210 16:30:55.774758       1 main.go:187] service host: https://10.255.240.1:443
I0210 16:30:55.775172       1 main.go:213] Testing communication with server
I0210 16:30:55.802572       1 main.go:218] Communication with server successful
I0210 16:30:55.802770       1 main.go:263] Active collectors: resourcequotas,replicasets,daemonsets,deployments,pods,nodes
I0210 16:30:55.802784       1 main.go:227] Starting metrics server: :8080
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x40218e]
goroutine 126 [running]:
panic(0x13249e0, 0xc420014030)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
main.(*deploymentCollector).collectDeployment(0xc42047a3a0, 0xc4202bf800, 0x0, 0x0, 0x0, 0x0, 0xc420257454, 0x6, 0x0, 0x0, ...)
	/home/brancz/go/src/k8s.io/kube-state-metrics/deployment.go:153 +0x27e
main.(*deploymentCollector).Collect(0xc42047a3a0, 0xc4202bf800)
	/home/brancz/go/src/k8s.io/kube-state-metrics/deployment.go:135 +0x133
k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc4209fdbd0, 0xc4202bf800, 0x1c77b80, 0xc42047a3a0)
	/home/brancz/go/src/k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:433 +0x63
created by k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/home/brancz/go/src/k8s.io/kube-state-metrics/vendor/github.com/prometheus/client_golang/prometheus/registry.go:434 +0x326
```

**Comment by stvnwrgs** [comment]
I get the same problem with your `quay.io/brancz/kube-state-metrics:v0.4.0` image.

**Comment by brancz** [comment]
Looks like we need a guard around `deployment.go:153`, for the time being you should be able to disable the deployments collector and it should work. You can specify collectors with the `-collectors` flag. So that would look like:

```
kube-state-metrics -collectors=resourcequotas,replicasets,daemonsets,pods,nodes
```

> Note I intentionally left out the deployment collector in that list

**Comment by brancz** [comment]
Fix is merged and `v0.4.1` is released.
